carbonmark/components/Dropdown/index.tsx

```diff
@@ -48,6 +48,11 @@ export function Dropdown<V, T extends FieldValues = FieldValues>(
     close();
   }, [selected]);
 
+  useEffect(() => {
+    if (initial === selected.id) return;
+    setSelected(defaultOption);
+  }, [initial]);
```

Comment:
It should already be set to defaultOption where useState is invoked. Why do we need this? It's not clear to me why.

I'm worried it will cause problems as it looks like it could trigger the other useEffect in some cases.

Reply:
Apologies, I should have left a comment around this to give some context. 

> Is it because the query params are parsed after first render?

Yeah, so the issue I was facing was - choose a sort option from the dropdown, reload the page. The sort query param remains as intended but the dropdown never gets updated and always takes the default sort value "recently-updated" - due to the query params not been parsed yet. 

So the idea here was to set the selected value on the dropdown if the sort value changes and exit if the value hadn't changed. Went down the rabbit hole with this one, so would love to hear other suggestions on this one if you have any - I've tried by waiting for the router.isReady to be true before rendering the dropdown (which works in theory - however throws hydration errors, same with not rendering the page until the router.isReady param is true).

Reply:
Hey @0xemc - thank you for this tip. I went down the rabbit hole on this one completely and this is a great simple solution.